### DISCUSSION TITLE
PEM SSL ERROR

### DISCUSSION THREAD

**Discussion by openaspace** [modoboa/modoboa#2648] [Help]
Hello I'm having this problems and no mail IN / OUT can't be sent.

Oct 16 12:37:28 Modoboa-Mail-Server postfix/smtpd[41124]: fatal: no SASL authentication mechanisms
Oct 16 13:07:30 Modoboa-Mail-Server postfix/smtpd[50079]: fatal: no SASL authentication mechanisms
Oct 17 05:10:04 static dovecot: imap-login: Error: Diffie-Hellman key exchange requested, but no DH parameters provided. Set ssl_dh=</path/to/dh.pem
Oct 17 05:10:09 static dovecot: imap-login: Error: Diffie-Hellman key exchange requested, but no DH parameters provided. Set ssl_dh=</path/to/dh.pem
Oct 17 05:10:16 static dovecot: imap-login: Error: Diffie-Hellman key exchange requested, but no DH parameters provided. Set ssl_dh=</path/to/dh.pem

**Comment by openaspace** [comment]
Which is the right settings file?
`root@static:/srv/modoboa/env/lib/python3.9/site-packages# find / -name "*settings.py*"
/srv/modoboa/env/lib/python3.9/site-packages/drf_spectacular/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa_radicale/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa_contacts/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa_amavis/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/rest_framework_simplejwt/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/django/conf/project_template/project_name/settings.py-tpl
/srv/modoboa/env/lib/python3.9/site-packages/django/conf/global_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/django/core/management/commands/diffsettings.py
/srv/modoboa/env/lib/python3.9/site-packages/rest_framework_nested/runtests/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/rest_framework/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/core/app_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/core/dev_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/core/commands/templates/settings.py.tpl
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/admin/app_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/limits/app_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/modoboa/test_settings.py
/srv/modoboa/env/lib/python3.9/site-packages/ckeditor_demo/settings.py
/srv/modoboa/env/lib/python3.9/site-packages/reportlab/rl_settings.py`

**Reply by Spitfireap** [reply]
Should be here `/srv/modoboa/instance/instance/settings.py`. Are you sure your installation did end correctly ?

**Reply by openaspace** [reply]
Yes no errors.
All works.. but the certificate doesn't works.

I have the mail server behind Nginx proxy manager , therefore 80 and 443 are passed by the proxy.... this can be the problem why modoboa doesn't generate correctly the ssl?

I can do it manually?

**Reply by Spitfireap** [reply]
did you use let's encrypt or self signed on install ?

**Reply by openaspace** [reply]
git clone https://github.com/modoboa/modoboa-installer
> cd modoboa-installer
> sudo ./run.py <your domain>

**Reply by Spitfireap** [reply]
self signed then.
What's the output of ``cat /etc/postfix/main.cf | grep sasl`` please

**Comment by openaspace** [comment]
**Today (after the night) the mail server was not receiving/sending any  email...
after the  reboot now is working.**

root@mail:~# cat /etc/postfix/main.cf | grep sasl
smtpd_sasl_type = dovecot
smtpd_sasl_path = private/auth
smtpd_sasl_auth_enable = yes
broken_sasl_auth_clients = yes
smtpd_sasl_security_options = noanonymous
    $smtp_sasl_auth_cache_name
    $lmtp_sasl_auth_cache_name
      permit_sasl_authenticated




Thank you for the support!

**Reply by openaspace** [reply]
Transcript of session follows.

 Out: 220 mail.myserver.com ESMTP
 In:  EHLO [100.103.183.58]
 Out: 250-mail.myserver.com
 Out: 250-PIPELINING
 Out: 250-SIZE 11534336
 Out: 250-ETRN
 Out: 250-STARTTLS
 Out: 250-ENHANCEDSTATUSCODES
 Out: 250-8BITMIME
 Out: 250-DSN
 Out: 250 CHUNKING
 In:  STARTTLS
 Out: 220 2.0.0 Ready to start TLS
 In:  EHLO [100.103.183.58]
 Out: 250-mail.myserver.com
 Out: 250-PIPELINING
 Out: 250-SIZE 11534336
 Out: 250-ETRN
 Out: 250-AUTH PLAIN LOGIN
 Out: 250-AUTH=PLAIN LOGIN
 Out: 250-ENHANCEDSTATUSCODES
 Out: 250-8BITMIME
 Out: 250-DSN
 Out: 250 CHUNKING
 In:  auth login
 Out: 334 VXNlcm5hbWU6
 In:  Z2l1c2VwcGVAamFzYW5nby5kaWdpdGFs
 Out: 334 UGFzc3dvcmQ6
 In:  NDE4NlZhbmRlcmdyYWY=
 Out: 235 2.7.0 Authentication successful
 In:  MAIL FROM:<myname@example.digital> SIZE=885
 Out: 250 2.1.0 Ok
 In:  RCPT TO:<name@myserver.com>
 **### Out: 451 4.3.0 Error: queue file write error**
 In:  QUIT
 Out: 221 2.0.0 Bye

**Reply by Spitfireap** [reply]
https://github.com/modoboa/modoboa/issues/2475

**Reply by openaspace** [reply]
> #2475
conf modified to: 

 $max_server = 2

**Reply by openaspace** [reply]
Trying to delete the domain **server.com** on the omodoboa **mail.server.com**:

(b"doveconf: Fatal: Error in configuration file /etc/dovecot/conf.d/10-ssl.conf line 14: ssl_key: Can't open file /etc/ssl/private/mail.serever.com.key: Permission denied\n")
